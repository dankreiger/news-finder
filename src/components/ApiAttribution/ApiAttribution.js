import React from 'react';
import { string } from 'prop-types';
import { Col, Row } from 'reactstrap';

const ApiAttribution = ({ text, href }) => (
  <Row>
    <Col>
      <p>
        <a href={href}>{text}</a>
      </p>
    </Col>
  </Row>
);

ApiAttribution.propTypes = {
  text: string,
  href: string
};

ApiAttribution.defaultProps = {
  href: '#'
};

export default ApiAttribution;
