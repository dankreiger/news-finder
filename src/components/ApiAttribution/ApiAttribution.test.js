import React from 'react';
import ApiAttribution from './ApiAttribution';
import { shallow } from 'enzyme';

describe('ApiAttribution', () => {
  let component;
  let anchor;
  const mockProps = {
    text: 'NewsApi',
    href: 'https://newsapi.org/'
  };

  beforeEach(() => {
    component = shallow(
      <ApiAttribution text={mockProps.text} href={mockProps.href} />
    );
    anchor = component.find('a');
  });

  it('renders without crashing', () => {
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('renders an anchor element', () => {
    expect(anchor.length).toBe(1);
  });

  it('sets components text props into the anchor element text', () => {
    expect(anchor.text()).toEqual(mockProps.text);
  });

  it('sets components text props into the anchor element text', () => {
    expect(anchor.prop('href')).toEqual(mockProps.href);
  });
});
