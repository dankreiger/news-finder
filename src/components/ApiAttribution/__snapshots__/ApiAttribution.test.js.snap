// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiAttribution renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApiAttribution
    href="https://newsapi.org/"
    text="NewsApi"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p>
          <a
            href="https://newsapi.org/"
          >
            NewsApi
          </a>
        </p>
      </Col>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          <a
            href="https://newsapi.org/"
          >
            NewsApi
          </a>
        </p>,
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://newsapi.org/"
          >
            NewsApi
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NewsApi",
            "href": "https://newsapi.org/",
          },
          "ref": null,
          "rendered": "NewsApi",
          "type": "a",
        },
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            <a
              href="https://newsapi.org/"
            >
              NewsApi
            </a>
          </p>
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <p>
            <a
              href="https://newsapi.org/"
            >
              NewsApi
            </a>
          </p>,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://newsapi.org/"
            >
              NewsApi
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NewsApi",
              "href": "https://newsapi.org/",
            },
            "ref": null,
            "rendered": "NewsApi",
            "type": "a",
          },
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
